client/state/data-layer/wpcom/comments/index.js

```diff
@@ -58,6 +117,16 @@ export const addComments = ( { dispatch }, { siteId, postId }, next, { comments,
 	}
 };
 
+export const writePostCommentSuccess = ( { dispatch, getState }, { siteId, postId, parentCommentId, placeholderId }, next, comment ) => {
+	// remove placeholder from state
+	dispatch( { type: COMMENTS_REMOVE, siteId, postId, commentId: placeholderId } );
+	// add new comment to state with updated values from server
+	dispatch( { type: COMMENTS_RECEIVE, siteId, postId, comments: [ comment ], skipSort: !! parentCommentId } );
+	// increment comments count
+	const commentCount = getPostTotalCommentsCount( getState(), siteId, postId );
+	dispatch( { type: COMMENTS_COUNT_RECEIVE, siteId, postId, totalCommentsCount: commentCount + 1 } );
```

Comment:
Changing this into an optimistic action (which we don't need to do at the moment) is an interesting thought. 

We can increment during `COMMENTS_WRITE`, and rollback in the error handler, but we might run into a case where `COMMENTS_COUNT_RECEIVE` is dispatched from someplace else while the comment is still pending.

I'm thinking we'd either have to add a little bit more state in that case to separate a pending from actual count, or dispatch another `COMMENT_COUNT` request in the success/error handler.